Source/WebCore/bindings/js/JSDOMPromiseDeferred.cpp

```diff
@@ -326,4 +326,13 @@ void DeferredPromise::handleUncaughtException(CatchScope& scope, JSDOMGlobalObje
     reportException(&lexicalGlobalObject, exception);
 };
 
+std::pair<Ref<DOMPromise>, Ref<DeferredPromise>> createPromiseAndWrapper(Document& document)
```

Comment:
[WebTransport.cpp](https://searchfox.org/wubkat/source/Source/WebCore/Modules/webtransport/WebTransport.cpp) also has a very similar version of this, in case you want to do more clean up.

Reply:
Thanks Tim, that makes sense indeed, will include that.